internal/blockstore/commit_and_query.go

```diff
@@ -477,9 +477,6 @@ func (s *Store) GetHeaderByHash(blockHash []byte) (*types.BlockHeader, error) {
 // DoesTxIDExist returns true if any of the committed block has a transaction with
 // the given txID. Otherwise, it returns false
 func (s *Store) DoesTxIDExist(txID string) (bool, error) {
-	s.mu.RLock()
-	defer s.mu.RUnlock()
-
```

Comment:
It is correct to remove this lock only because this method is used optimistically from the front end. Lets ass a comment that specifies that, such that it is never used in the validation & commit path.

Reply:
It is safe to remove the locks because this method calls a function of one of its internal classes, which is protected.